fah_alchemy/storage/statestore.py

```diff
@@ -1237,3 +1237,50 @@ def remove_credentialed_identity(
 
         with self.transaction() as tx:
             tx.run(q)
+
+    def list_scopes(
+        self, identifier: str, cls: type[CredentialedEntity]
+    ) -> List[Scope]:
+        """List all scopes for which the given entity has access."""
+        q = f"""
+        MATCH (n:{cls.__name__} {{identifier: '{identifier}'}})
+        MERGE (n)-[r:HAS_SCOPE]->(s:Scope)
+        RETURN s
+        """
+
+        with self.transaction() as tx:
+            res = tx.run(q)
+
+        scopes = []
+        for record in res:
+            scope_rec = record["s"]
+            # use Scope.from_str_tuple to ensure that the scope wildcards are
+            # deserialized properly ie that they are converted from "*" to None
+            scope = Scope.from_str_tuple(
+                (scope_rec["org"], scope_rec["campaign"], scope_rec["project"])
+            )
+            scopes.append(scope)
+        return scopes
+
+    def add_scope(self, identifier: str, cls: type[CredentialedEntity], scope: Scope):
```

Comment:
Instead of Scopes as nodes, let's just add to the `"scopes"` property of the identity node. You'll need to add the given scope to the existing list of properties using e.g.: https://stackoverflow.com/questions/21979782/how-to-push-values-to-property-array-in-cypher#21980719

Reply:
Hopefully addressed this. :)